vortex-sampling-compressor/src/lib.rs

```diff
@@ -242,13 +243,17 @@ impl<'a> SamplingCompressor<'a> {
             return Ok(CompressedArray::uncompressed(arr.clone()));
         }
 
+        // It's worth computing pruning stats eagerly
+        compute_pruning_stats(arr)?;
```

Comment:
Maybe we should do it upon writing, not during compression?

Reply:
the reason to do it during compression is that many encodings don't implement stats... so if we do it here, we'll get the stats and propagate them. whereas at write time, we won't get them at all (unless we canonicalize to compute stats, which would be dumb)

but this is actually too aggressive, slows down write by 2x in the benchmarks. we don't need this on every array, just on column chunks, so will just call this fn on children in the chunked and struct compressors

Reply:
well... might be a good time to implement stats but won't block